src/ecpool_sup.erl

```diff
@@ -90,3 +111,49 @@ pool_spec(Pool, Mod, Opts) ->
 
 child_id(Pool) -> {pool_sup, Pool}.
 
+%%--------------------------------------------------------------------
+%% Internal functions
+%%--------------------------------------------------------------------
+
+kill_ecpool_pool_sup_if_stuck() ->
+    case process_info(whereis(?MODULE), links) of
+        {links, LinkedPids} ->
```

Comment:
this going to kill it's supervisor too ?

Reply:
No it won't, the linked process (`ecpool_pool_sup`) has not finished it initialization so that has not been supervised by `ecpool_sup`.